src/coreclr/jit/indirectcalltransformer.cpp

```diff
@@ -821,8 +837,14 @@ class IndirectCallTransformer
                 call->gtCallType                = CT_USER_FUNC;
                 call->gtCallMoreFlags |= GTF_CALL_M_DEVIRTUALIZED;
                 call->gtCallMoreFlags &= ~GTF_CALL_M_DELEGATE_INV;
-                // TODO-GDV: To support R2R we need to get the entry point
-                // here. We should unify with the tail of impDevirtualizeCall.
+
+#ifdef FEATURE_READYTORUN
+                // We should have set this in CreateCheck.
+                assert(m_r2rEntryPoint.accessType != IAT_VALUE || m_r2rEntryPoint.addr != nullptr);
+                call->setEntryPoint(m_r2rEntryPoint);
```

Comment:
I should probably look into implementing a limited version of `getFunctionEntryPoint` too so we can get "normal" R2R entry point on the direct call node, which should have better steady-state performance in case we do not manage to inline it.